TheJumpCloud/jcapi-powershell
# CUT-2378: -Attribues Object needs custom type

## Issues
* [CUT-2378](https://jumpcloud.atlassian.net/browse/CUT-2378) - -Attribues Object needs custom type

## What does this solve?
Through some discovery it was determined that the -Attributes object is being flattened by the SDKs and does not necessarily need a custom type. 

As an example the following script would set sudo to enabled and withoutPassword to false
```powershell
$attributes = [JumpCloud.SDK.V2.Models.IGraphOperationSystemAttributes]@{ 'sudo' = @{'enabled' = $true; 'withoutPassword' = $false } }
Set-JcSdkSystemAssociation -SystemId 61982961719e942987898f39 -id 5ef5fe69a899bd46963ab24a -op 'update' -type 'user' -Attributes $attributes
```

We can achieve the same result by doing the following:
```powershell
Set-JcSdkSystemAssociation -SystemId 61982961719e942987898f39 -id 5ef5fe69a899bd46963ab24a -op 'update' -type 'user' -Attributes @{ 'sudoEnabled' = $true; 'SudoWithoutPassword' = $true }
```

This is due to the SDKs doing a sort of flattening to the sudo object and allowing us to instead call the sudoEnabled or SudoWithoutPassword property to set those values.

Rather than creating a custom object, we can update the documentation for this functionality

This behavior can also be applied to the `Set-JcSdkUserAssociation` function

## Is there anything particularly tricky?
N/A

## How should this be tested?

Pull the branch and import the updated version of the V2 SDK
Attempt to run the following script and ensure that the results are updated in the console:
```powershell
Set-JcSdkSystemAssociation -SystemId <SystemID> -id <UserID> -op 'update' -type 'user' -Attributes @{ 'sudoEnabled' = $true; 'SudoWithoutPassword' = $true }
```

[CUT-2378]: https://jumpcloud.atlassian.net/browse/CUT-2378?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
